<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>504</width>
    <height>401</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>N-Puzzle Solver</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>321</width>
      <height>271</height>
     </rect>
    </property>
    <property name="title">
     <string>Puzzle State</string>
    </property>
    <widget class="QPlainTextEdit" name="edit_puzzleState">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>300</width>
       <height>240</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Courier New</family>
       <pointsize>14</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="cursor" stdset="0">
      <cursorShape>IBeamCursor</cursorShape>
     </property>
     <property name="plainText">
      <string>Generate a puzzle..</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>20</y>
      <width>141</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Configuration</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>121</width>
       <height>54</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="button_generatePuzzle">
        <property name="text">
         <string>Generate Puzzle</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>121</width>
       <height>138</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QComboBox" name="combo_chooseMethod">
        <item>
         <property name="text">
          <string>BFS</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>DFS</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>A* Search H1</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>A* Search H2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>A* Search H3</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Iterative Deeping</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_start">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(0, 170, 0);
</string>
        </property>
        <property name="text">
         <string>Start</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_step">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 170, 0);</string>
        </property>
        <property name="text">
         <string>Single Step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_showTheSol">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 0);</string>
        </property>
        <property name="text">
         <string>Show the Solution</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_cancel">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 0, 0);</string>
        </property>
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>261</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Details</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>188</width>
       <height>24</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Iteration Number</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="edit_iterationNumber">
        <property name="text">
         <string>0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>22</x>
       <y>50</y>
       <width>82</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Steps to Solution</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="edit_step2sol">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>50</y>
       <width>98</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>504</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>

